example_configs/artemis-2.yml

```diff
@@ -0,0 +1,25 @@
+---
+lowercaseOutputName: true
+lowercaseOutputLabelNames: true
+rules:
+  - pattern: "^org.apache.activemq.artemis<broker=\"([^\"]*)\"><>([^:]*):\\s(.*)"
+    attrNameSnakeCase: true
+    name: artemis_$2
+    type: COUNTER
+    labels:
+        host: $1
```

Comment:
A host label is unusual. Can you share example output?

Reply:
Sure:
```
artemis_message_count{address="ExpiryQueue",host="infra1-live",} 0.0
artemis_message_count{queue="ExpiryQueue",address="ExpiryQueue",host="infra1-live",} 0.0
artemis_message_count{queue="DLQ",address="DLQ",host="infra1-live",} 0.0
artemis_message_count{queue="notif.cf762ce6-f5df-11e7-ab73-0050568f7196.ActiveMQServerImpl_serverUUID=71f2d8ca-e3f6-11e7-9700-0050568f7196",address="activemq.notifications",host="infra1-live",} 0.0
artemis_message_count{address="DLQ",host="infra1-live",} 0.0
artemis_message_count{address="activemq.notifications",host="infra1-live",} 0.0
artemis_message_count{queue="$.artemis.internal.sf.example-cluster.71f2d8ca-e3f6-11e7-9700-0050568f7196",address="$.artemis.internal.sf.example-cluster.71f2d8ca-e3f6-11e7-9700-0050568f7196",host="infra1-live",} 0.0
artemis_message_count{address="$.artemis.internal.sf.example-cluster.71f2d8ca-e3f6-11e7-9700-0050568f7196",host="infra1-live",} 0.0
```
This is the output of one Artemis broker. The `host` label is the broker name (could be renamed to that if you want).

Reply:
Is the broker name the broker that we're talking to?

Reply:
yes

Reply:
That should be removed then, as that's already covered by the instance label.

Reply:
The instance label is IP address and port which is hard to identify when several instances are running on the same machine. The broker name does not need to be the host name. It can be configured for each Artemis broker instance.

Reply:
That's still a matter for service discovery on Prometheus's end.

Reply:
Sorry, I'm no expert for Prometheus. How would I lookup a specific broker without having it's name as label?